controller/consoleapplication_controller.go

```diff
@@ -0,0 +1,131 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	gitservice "github.com/openshift-console/console-application-operator/pkg/git-service"
+)
+
+// ConsoleApplicationReconciler reconciles a ConsoleApplication object
+type ConsoleApplicationReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *ConsoleApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling ConsoleApplication...")
+
+	// Retrieving the ConsoleApplication CR instance requested for reconciliation
+	operatorCR := &appsv1alpha1.ConsoleApplication{}
+	if err := r.Get(ctx, req.NamespacedName, operatorCR); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Error(err, "ConsoleApplication Not Found")
+			return NoRequeue()
+		}
+		logger.Error(err, "Unable to fetch ConsoleApplication CR")
+		SetDegraded(operatorCR, appsv1alpha1.ReasonOperatorResourceNotAvailable.String(), err.Error())
+		if err := r.Status().Update(ctx, operatorCR); err != nil {
+			return RequeueWithError(err)
+		}
+		return RequeueOnError(err)
+	}
+
+	init := operatorCR.Status.Conditions == nil
+	if init {
+		operatorCR.Status.Conditions = make([]metav1.Condition, 0)
+		SetStarted(operatorCR)
+		if err := r.Status().Update(ctx, operatorCR); err != nil {
+			return RequeueWithError(err)
+		}
+	}
+
+	// Fetching the secret resource if specified in the CR
+	secretResourceName := operatorCR.Spec.Git.SourceSecretRef
+	decodedSecret := ""
+	if secretResourceName != "" {
+		secret := &corev1.Secret{}
+		if err := r.Get(ctx, client.ObjectKey{
+			Namespace: req.Namespace,
+			Name:      secretResourceName,
+		}, secret); err != nil {
+			logger.Error(err, "Unable To Find Secret Resource")
+			SetFailed(operatorCR, appsv1alpha1.ReasonSecretResourceNotFound.String(), err.Error())
+			if err := r.Status().Update(ctx, operatorCR); err != nil {
+				return RequeueWithError(err)
+			}
+			return NoRequeue()
+		}
+		// Decoding the secret data - Considering the Basic Auth secret
+		decodedSecret = string(secret.Data["password"])
+		logger.Info("Secret Decoded: " + decodedSecret)
+	}
+
+	// Checking if the Git Repository is reachable
+	gs := gitservice.New(operatorCR.Spec.Git.Url, operatorCR.Spec.Git.Reference, decodedSecret, logger)
+	g_status, g_reason := gs.IsRepoReachable()
```

Comment:
let's try and be consistent with using camelCase.
```suggestion
	gStatus, gReason := gs.IsRepoReachable()
```